gcsfs/core.py

```diff
@@ -1824,6 +1824,9 @@ def __init__(
         self.acl = acl
         self.checker = get_consistency_checker(consistency)
 
+        if "a" in self.mode:
+            warnings.warn("Append mode 'a' is not supported in GCS. Using overwrite mode instead.")
```

Comment:
Should actually replace the mode, no?
```suggestion
            warnings.warn("Append mode 'a' is not supported in GCS. Using overwrite mode instead.")
            self.mode = self.mode.replace("a", "w")
```